leptonai/cli/pod.py

```diff
@@ -84,24 +87,42 @@ def pod():
     help="Secrets to use for pulling images.",
     multiple=True,
 )
+@click.option(
+    "--node-group",
+    "-ng",
+    "node_groups",
+    help="Node group for the job. If not set, use on-demand resources.",
```

Comment:
nit: add "you can repeat this flag multiple times to choose multiple node groups."